src/register-org/components/organisation-services-access/organisation-services-access.component.ts

```diff
@@ -15,65 +12,58 @@ import { RegisterOrgService } from '../../services/register-org.service';
 })
 export class OrganisationServicesAccessComponent extends RegisterComponent implements OnInit, OnDestroy {
   public readonly CATEGORY_SERVICE_ACCESS = 'Service';
+  public readonly SERVICE_NOT_LISTED = 'ServiceNotListed';
   public servicesFormGroup: FormGroup;
-  public lovRefDataSubscription: Subscription;
-  public services: LovRefDataModel[] = [];
+  public services: OrganisationService[];
   public selectedServices: string[] = [];
   public validationErrors: { id: string, message: string }[] = [];
+  public noServicesError: string;
+  public otherServicesError: string;
+  public showOtherServicesInput: boolean;
 
   constructor(public readonly router: Router,
-    public readonly registerOrgService: RegisterOrgService,
-    private readonly lovRefDataService: LovRefDataService,
-    private readonly route: ActivatedRoute
+    public readonly registerOrgService: RegisterOrgService
   ) {
     super(router, registerOrgService);
   }
 
   public ngOnInit(): void {
     super.ngOnInit();
 
-    this.services = SERVICES_REF_DATA;
-    // TODO: Integration with ref data
-    //  1. Delete the above line where it uses the mock data
-    //  2. Uncommented the below lines to test service error
-    const realApi = this.route.snapshot.queryParams.api;
-    if (realApi) {
-      this.lovRefDataService.getListOfValues(this.CATEGORY_SERVICE_ACCESS, false).subscribe((lov) => {
-        this.services = lov;
-      // eslint-disable-next-line @typescript-eslint/no-unused-vars
-      }, (error) => {
-        this.router.navigate([this.registerOrgService.REGISTER_ORG_NEW_ROUTE, 'service-down']);
-      });
-    }
+    this.services = ORGANISATION_SERVICES;
 
     this.servicesFormGroup = new FormGroup({
-      services: new FormControl(null)
+      services: new FormArray([]),
+      otherServices: new FormControl(null)
     });
 
     this.setFormControlValues();
   }
 
   public ngOnDestroy(): void {
     super.ngOnDestroy();
-
-    if (this.lovRefDataSubscription) {
-      this.lovRefDataSubscription.unsubscribe();
-    }
   }
 
   public onServicesSelectionChange(event: any): void {
     if (event.target.checked) {
-      this.services.find((service) => service.value_en === event.target.value).selected = true;
+      this.selectedServices.push(event.target.value);
     } else {
-      this.services.find((service) => service.value_en === event.target.value).selected = false;
+      const serviceIndex = this.selectedServices.findIndex((service) => service === event.target.value);
```

Comment:
@johnbenjamin-hmcts - does this still work if `findIndex()` returns -1?

Reply:
@Daniel-Lam `ORGANISATION_SERVICES` is a hard-coded master list of services that we display in the page, so the `selectedServices` must always contain the services from this master list.